lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,41 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      final titleString = nodes[0].text;
+      RegExp regex = RegExp(r'\b\d+\b');
+      Iterable<RegExpMatch> matches = regex.allMatches(titleString);
+      List<String> numbers = matches.map((match) => match.group(0)).toList();
```

Comment:
```suggestion
      final RegExp regex = RegExp(r'\b\d+\b');
      final Iterable<RegExpMatch> matches = regex.allMatches(titleString);
      final List<String> numbers = matches.map((match) => match.group(0)).toList();
```

Suggest always consider using constants.
If using compile-time constants `const` is impossible, use runtime constants `final`.

This helps enhance code maintainability, performance, and some common-seen issues such as memory re-assign problem, etc.

Reply:
[5e28563](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/5e2856363a20e9493fe777af17f61095c57c2fff)